src/bindings/c/src/ov_model.cpp

```diff
@@ -71,6 +71,38 @@ ov_status_e ov_model_input_by_index(const ov_model_t* model, const size_t index,
     return ov_status_e::OK;
 }
 
+ov_status_e ov_model_output_by_index(const ov_model_t* model,
+                                     const size_t index,
+                                     ov_output_const_node_t** output_node) {
+    if (!model || !output_node) {
+        return ov_status_e::INVALID_C_PARAM;
+    }
+    try {
+        auto result = std::const_pointer_cast<const ov::Model>(model->object)->output(index);
```

Comment:
Don't use std::const_pointer_cast here, we can define new C structure for ov::Output<ov::Node>

Reply:
Update it in new commit.